mailbox/api/src/main/java/org/apache/james/mailbox/model/FetchGroup.java

```diff
@@ -0,0 +1,151 @@
+
+/**
+ * Indicates the results fetched.
+ */
+public class FetchGroup {
+    /**
+     * For example: could have best performance when doing store and then
+     * forget. UIDs are always returned
+     */
+    public static final int NO_MASK = 0;
+    public static final int MINIMAL_MASK = 0x00;
+    public static final int MIME_DESCRIPTOR_MASK = 0x01;
+    public static final int HEADERS_MASK = 0x100;
+    public static final int FULL_CONTENT_MASK = 0x200;
+    public static final int BODY_CONTENT_MASK = 0x400;
+    public static final int MIME_HEADERS_MASK = 0x800;
+    public static final int MIME_CONTENT_MASK = 0x1000;
+
+    public static final FetchGroup MINIMAL = new FetchGroup(MINIMAL_MASK);
+    public static final FetchGroup HEADERS = new FetchGroup(HEADERS_MASK);
+    public static final FetchGroup FULL_CONTENT = new FetchGroup(FULL_CONTENT_MASK);
+    public static final FetchGroup BODY_CONTENT = new FetchGroup(BODY_CONTENT_MASK);
+
+    private final int content;
+    private final ImmutableSet<PartContentDescriptor> partContentDescriptors;
+
+    @VisibleForTesting
+    FetchGroup(int content) {
+        this(content, ImmutableSet.of());
+    }
+
+    @VisibleForTesting
+    FetchGroup(int content, ImmutableSet<PartContentDescriptor> partContentDescriptors) {
+        this.content = content;
+        this.partContentDescriptors = partContentDescriptors;
+    }
+
+    /**
+     * Contents to be fetched. Composed bitwise.
+     *
+     * @return bitwise description
+     * @see #MINIMAL_MASK
+     * @see #MIME_DESCRIPTOR_MASK
+     * @see #HEADERS_MASK
+     * @see #FULL_CONTENT_MASK
+     * @see #BODY_CONTENT_MASK
+     * @see #MIME_HEADERS_MASK
+     * @see #MIME_CONTENT_MASK
+     */
+    public int content() {
+        return content;
+    }
+
+    public FetchGroup or(int content) {
+         return new FetchGroup(this.content | content, partContentDescriptors);
+    }
+
+    /**
+     * Gets contents to be fetched for contained parts. For each part to be
+     * contained, only one descriptor should be contained.
+     *
+     * @return <code>Set</code> of {@link PartContentDescriptor}, or null if
+     *         there is no part content to be fetched
+     */
+    public Set<PartContentDescriptor> getPartContentDescriptors() {
+        return partContentDescriptors;
+    }
+
+    /**
+     * Adds content for the particular part.
+     * 
+     * @param path
+     *            <code>MimePath</code>, not null
+     * @param content
+     *            bitwise content constant
+     */
+    public FetchGroup addPartContent(MimePath path, int content) {
+        PartContentDescriptor newContent = partContentDescriptors.stream()
+            .filter(descriptor -> path.equals(descriptor.path()))
+            .findFirst()
+            .orElse(new PartContentDescriptor(path))
+            .or(content);
+
+        return new FetchGroup(this.content, ImmutableSet.<PartContentDescriptor>builder()
```

Comment:
```suggestion
        return new FetchGroup(this.content, 
        Stream.concat(
        partContentDescriptors.stream()
                .filter(descriptor -> descriptor.path().equals(path)),
                Stream.of(newContent))
                .collect(Guavate.toImmutableSet());
```

Reply:
No we need to update the stored partContent with or.